wordpress-mobile/WordPress-Android
# Feature/updated locale picker

This HW PR updates the look of the Application Locale picker. It's based on the Site Timezone picker.

You can filter by both label and localized label.

The original idea was to also provide locale suggestions, but because of the HW time constraints, I left it and some other potential improvements (getting rid of Pair and Triple) for the next HW.

Apologies for the size of the PR, but at the time feature flag seemed a bit of an overkill for the HW feature.

[![Image from Gyazo](https://i.gyazo.com/ef74e9dab13e3dbd1fab40555d8e5b04.png)](https://gyazo.com/ef74e9dab13e3dbd1fab40555d8e5b04)

To test:
- Navigate to App Settings, and open the Interface Language option.
- Confirm that the new locale picker works as expected:
      - You can filter locales by both their label and localized label (eg. typing "Dani" and "Dans" should produce list with "Danish/Dansk" locale)
      - Tapping on locale changes app language.
      - Current app locale is displayed a the top of the list. Tapping on it closes the list.

## Regression Notes
1. Potential unintended areas of impact
- Locale Picker is not working!

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing + VM tests.

3. What automated tests I added (or what prevented me from doing so)
- Hopefully all of them!

PR submission checklist:
